eng/targets/Settings.props

```diff
@@ -18,6 +18,7 @@
     <ToolsetPackagesDir>$(RepoRoot)build\ToolsetPackages\</ToolsetPackagesDir>
 
     <RoslynPortableTargetFrameworks>net472;netcoreapp2.1</RoslynPortableTargetFrameworks>
```

Comment:
Why not just reverse this variable instead? (I assume I'm going to regret asking this question.)

Reply:
Doesn't affect just the tests. I'm not opposed to it, just not what this change was intended to do.

Reply:
What else cares about it besides the test tooling?

Reply:
The target frameworks themselves may be different for executables and tests. For instance, we may want to move the tests to target netcoreapp50 and keep the executables on 3.1

Reply:
That's not now, obviously, but I wanted to plan for future changes.

Reply:
> The target frameworks themselves may be different for executables and tests

This is true however the way we generally handle it is by special casing that test project and having a custom `<TargetFrameworks>`. These exceptions exist today because of `netcoreapp3.0` and DIM. I would expect them to exist again in future releases. 

> I'm not opposed to it, just not what this change was intended to do.

I 100% think we should just be swapping out the order here vs. adding a new property. There is no reason to have a different order here for test and non-test projects. The only real impact the order has here is changing the default tests that are run which is the entire point of this PR.